apps/web/app/(use-page-wrapper)/workflow/new/page.tsx

```diff
@@ -116,6 +118,42 @@ const Page = async ({ searchParams }: PageProps) => {
       },
     });
 
+    if (parsedEventTypeId) {
+      const permissionService = new PermissionCheckService();
+      let hasPermission = false;
+
+      if (user.org?.id) {
+        hasPermission = await permissionService.checkPermission({
+          userId,
+          teamId: user.org?.id,
+          permission: "eventType.update",
+          fallbackRoles: [MembershipRole.ADMIN, MembershipRole.OWNER],
+        });
+      } else if (parsedTeamId) {
+        hasPermission = await permissionService.checkPermission({
+          userId,
+          teamId: parsedTeamId,
+          permission: "eventType.update",
+          fallbackRoles: [MembershipRole.ADMIN, MembershipRole.OWNER],
+        });
+      } else {
+        const eventTypeRepo = new EventTypeRepository(prisma);
+        hasPermission = await eventTypeRepo.isPersonalUserEventType({
+          userId,
+          eventTypeId: parsedEventTypeId,
+        });
+      }
+
+      if (hasPermission) {
```

Comment:
What happens if `hasPermission` is false? We should be handling that and showing the error message to the user.